app/controllers/user_profiles_controller.rb

```diff
@@ -25,6 +27,15 @@ def update
     end
   end
 
+  def student_stats_data
```

Comment:
I suggest using a single endpoint for all the user profile stats. You can structure the jbuilder view to separate student vs instructor stats, if you need to. But just having one API call and one path through the React data flow will make this easier to work with and reason about.